ct.sh

```diff
@@ -9,28 +9,41 @@ if [ -z $TRAVIS_RUST_VERSION ]; then
     exit 1
 fi
 
+export CFLAGS_x86_64_fortanix_unknown_sgx="-isystem/usr/include/x86_64-linux-gnu -mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening"
+export CC_x86_64_fortanix_unknown_sgx=clang-11
+
 if [ $TRAVIS_RUST_VERSION = "stable" ] || [ $TRAVIS_RUST_VERSION = "beta" ] || [ $TRAVIS_RUST_VERSION = "nightly" ]; then
     rustup default $TRAVIS_RUST_VERSION
-    # make sure that explicitly providing the default target works
-    cargo test --target x86_64-unknown-linux-gnu
-    cargo test --release
-    cargo test --features spin_threading
-    cargo test --features rust_threading
-    cargo test --features custom_time,custom_gmtime_r
-    cargo test --features zlib
-    cargo test --features pkcs12
-    cargo test --features pkcs12_rc2
-    cargo test --features force_aesni_support
-    cargo test --features dsa
+    rustup target add --toolchain $TRAVIS_RUST_VERSION $TARGET
+    # The SGX target cannot be run under test like a ELF binary
+    if [ "$TARGET" != "x86_64-fortanix-unknown-sgx" ]; then
+        # make sure that explicitly providing the default target works
+        cargo test --target $TARGET --release
+        cargo test --features pkcs12 --target $TARGET
+        cargo test --features pkcs12_rc2 --target $TARGET
+        cargo test --features dsa --target $TARGET
+        cargo test --features spin_threading --target $TARGET
+        cargo test --features rust_threading --target $TARGET
+        cargo test --features custom_time,custom_gmtime_r --target $TARGET
+        # test the async support
+        cargo test --test async_session --features=std,threading,tokio,tokio/net,tokio/io-util,tokio/macros,tokio/rt --target $TARGET
```

Comment:
all the tokio features can be replaced with `tokio/full`

Reply:
nice catch!